samples/spring-transfer-workflow/src/main/java/com/example/transfer/TransferState.java

```diff
@@ -0,0 +1,43 @@
+package com.example.transfer;
+
+import static com.example.transfer.TransferState.TransferStatus.*;
+
+// tag::domain[]
+public record TransferState(Transfer transfer, TransferStatus status) {
+
+  public record Transfer(String from, String to, int amount) { // <1>
+  }
+
+  public enum TransferStatus { // <2>
+    STARTED, MANUAL_APPROVAL_REQUIRED, ACCEPTED, SUCCESSFUL_WITHDRAWAL, COMPLETED, REJECTED
```

Comment:
I would reduce the status. 

To demo `pause` we can require a second `sign-off`. The state can have the name of the initiator and then the name of the second person that authorizes the transfer, for instance.